src/nyx/features/ngldm.cpp

```diff
@@ -0,0 +1,346 @@
+#include <limits.h>
+#include "ngldm.h"
+#include "../environment.h"
+
+NGLDMfeature::NGLDMfeature() : FeatureMethod("NGLDMfeature")
+{
+	provide_features ({
+		NGLDM_LDE,
+		NGLDM_HDE,
+		NGLDM_LGLCE,
+		NGLDM_HGLCE,
+		NGLDM_LDLGLE,
+		NGLDM_LDHGLE,
+		NGLDM_HDLGLE,
+		NGLDM_HDHGLE,
+		NGLDM_GLNU,
+		NGLDM_GLNUN,
+		NGLDM_DCNU,
+		NGLDM_DCNUN,
+		NGLDM_GLM,
+		NGLDM_GLV,
+		NGLDM_DCM,
+		NGLDM_DCV,
+		NGLDM_DCE,
+		NGLDM_DCENE });
+}
+
+void NGLDMfeature::clear_buffers()
+{
+	f_LDE =
+		f_HDE =
+		f_LGLCE =
+		f_HGLCE =
+		f_LDLGLE =
+		f_LDHGLE =
+		f_HDLGLE =
+		f_HDHGLE =
+		f_GLNU =
+		f_GLNUN =
+		f_DCNU =
+		f_DCNUN =
+		f_GLCM =
+		f_GLV =
+		f_DCM =
+		f_DCV =
+		f_DCE =
+		f_DCENE = 0;
+}
+
+template <class PixelCloud> void NGLDMfeature::gather_unique_intensities (std::vector<PixIntens> & V, PixelCloud & C, PixIntens max_inten)
+{
+	std::unordered_set<PixIntens> U;
+	PixIntens range = max_inten - 0;
+	unsigned int nGrays = theEnvironment.get_coarse_gray_depth();
+	for (Pixel2 p : C)
+	{
+		PixIntens inten_ = Nyxus::to_grayscale(p.inten, 0, range, nGrays, Environment::ibsi_compliance);
+		U.insert(inten_);
+	}
+
+	// -- Set to vector to be able to know each intensity's index
+	V.insert (V.end(), U.begin(), U.end()); //std::vector<PixIntens> V (U.begin(), U.end());
+	std::sort (V.begin(), V.end());
+}
+
+template <class Imgmatrix> void NGLDMfeature::calc_ngldm (SimpleMatrix<unsigned int> & NGLDM, Imgmatrix & I, std::vector<PixIntens> & V, PixIntens max_inten)

```

Comment:
should we add `const` qualifier to things that we take by reference but do not change in the function?